frontend/src/tree/TreeItemContextMenu.tsx

```diff
@@ -44,60 +73,109 @@ export const TreeItemContextMenu = ({
-          return element;
-        })}
-      {item.editable ? (
-        <MenuItem
-          key="rename"
-          onClick={enterEditingMode}
-          data-testid="rename-tree-item"
-          disabled={readOnly}
-          aria-disabled
-        >
-          <ListItemIcon>
-            <EditIcon fontSize="small" />
-          </ListItemIcon>
-          <ListItemText primary="Rename" />
-        </MenuItem>
-      ) : null}
-      {item.deletable ? (
-        <MenuItem key="delete-tree-item" onClick={deleteItem} data-testid="delete" disabled={readOnly} aria-disabled>
-          <ListItemIcon>
-            <DeleteIcon fontSize="small" />
-          </ListItemIcon>
-          <ListItemText primary="Delete" />
-        </MenuItem>
-      ) : null}
-    </Menu>
+    <>
+      <Menu
+        id="treeitem-contextmenu"
+        anchorEl={menuAnchor}
+        keepMounted
+        open
+        onClose={onClose}
+        data-testid="treeitem-contextmenu"
+        disableRestoreFocus={true}
+        getContentAnchorEl={null}
+        anchorOrigin={{
+          vertical: 'bottom',
+          horizontal: 'right',
+        }}
+      >
+        {treeItemContextMenuContributions
+          .filter((contribution) => contribution.props.canHandle(item))
+          .map((contribution, index) => {
+            const props: TreeItemContextMenuComponentProps = {
+              editingContextId,
+              item,
+              readOnly,
+              onClose,
+              selection,
+              setSelection,
+              expandItem,
+              key: index.toString(),
+              treeId: treeId,
+            };
+            const element = React.createElement(contribution.props.component, props);
+            return element;
+          })}
+        {item.editable ? (
+          <MenuItem
+            key="rename"
+            onClick={enterEditingMode}
+            data-testid="rename-tree-item"
+            disabled={readOnly}
+            aria-disabled
+          >
+            <ListItemIcon>
+              <EditIcon fontSize="small" />
+            </ListItemIcon>
+            <ListItemText primary="Rename" />
+          </MenuItem>
+        ) : null}
+        {item.deletable ? (
+          <MenuItem key="delete-tree-item" onClick={deleteItem} data-testid="delete" disabled={readOnly} aria-disabled>
+            <ListItemIcon>
+              <DeleteIcon fontSize="small" />
+            </ListItemIcon>
+            <ListItemText primary="Delete" />
+          </MenuItem>
+        ) : null}
+      </Menu>
+      <Snackbar
+        anchorOrigin={{
+          vertical: 'bottom',
+          horizontal: 'right',
+        }}
+        open={state.message != null}
+        autoHideDuration={3000}
+        onClose={() => closeToast}
+        message={state.message}
+        action={
+          <IconButton size="small" aria-label="close" color="inherit" onClick={() => closeToast}>
```

Comment:
`closeToast` is a function so the code should either be `onClick={closeToast}` or `{onClick=() => closeToast()}`. This does not work.